app/src/main/java/org/kiwix/kiwixmobile/history/HistoryActivity.java

```diff
@@ -111,15 +116,32 @@ protected void onCreate(Bundle savedInstanceState) {
       actionBar.setTitle(R.string.history);
     }
 
-
     historyAdapter = new HistoryAdapter(historyList, deleteList, this);
     recyclerView.setAdapter(historyAdapter);
+    layoutManager = new LinearLayoutManager(this, RecyclerView.VERTICAL, false);
+    recyclerView.setLayoutManager(layoutManager);
   }
 
   @Override
   protected void onResume() {
     super.onResume();
     presenter.loadHistory(sharedPreferenceUtil.getShowHistoryCurrentBook());
+    if (listState != null) {
+      layoutManager.onRestoreInstanceState(listState);
```

Comment:
I think we can move this line to `onRestoreInstanceState(Bundle state)` of this activity. In that case, `listState` can be removed. Correct me if I'm wrong.

Reply:
@abdulwd  I checked the documentation and I think `listState` should be in `onResume()` only since `onResume` is called after `onRestoreInstanceState()`, which is not always called.
>>Most implementations will simply use onCreate(Bundle) to restore their state, but it is sometimes convenient to do it here after all of the initialization has been done or to allow subclasses to decide whether to use your default implementation. The default implementation of this method performs a restore of any view state that had previously been frozen by onSaveInstanceState(Bundle)

Here is the link https://developer.android.com/reference/android/app/Activity.html#onRestoreInstanceState(android.os.Bundle)